components/map/index.jsx

```diff
@@ -0,0 +1,162 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// Implements the [Map design pattern](https://lightningdesignsystem.com/components/map/) in React.
+// Based on SLDS v2.4.0
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+// ### shortid
+// [npmjs.com/package/shortid](https://www.npmjs.com/package/shortid)
+// shortid is a short, non-sequential, url-friendly, unique id generator
+import shortid from 'shortid';
+
+import Icon from '../icon';
+import Modal from '../modal';
+import { MAP } from '../../utilities/constants';
+
+const displayName = MAP;
+
+const propTypes = {
+	/**
+	 * CSS class names to be added to the container element. `array`, `object`, or `string` are accepted.
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+	/**
+	 * HTML id for component.
+	 */
+	id: PropTypes.string,
+	/**
+	 * Whether the modal containing the map is open. Only for `modal` variant
+	 */
+	isModalOpen: PropTypes.bool,
```

Comment:
`isOpen` should be controlled another way. See comments about composition with `Modal` below.